types/test/types/rpc.ts

```diff
@@ -440,8 +440,14 @@ export default <ExportedHandler<Env>>{
       expectTypeOf(s.objectProperty.z(false)).toEqualTypeOf<Promise<number>>(); // (pipelining)
 
       expectTypeOf(s.everySerializable).not.toBeNever();
-      expectTypeOf(s.nonSerializable1).toBeNever();
-      expectTypeOf(s.nonSerializable2).returns.toBeNever();
+
+      // These are _not_ serializable, but will nevertheless not be typed as `never`. This
+      // is because TS can't distinguish between a class instance and an interface. At runtime,
+      // using an RPC method like this will throw:
+      //   "Could not serialize object. This type does not support serialization.""
+      expectTypeOf(s.nonSerializable1).not.toBeNever();
```

Comment:
These new lines of test aren't actually testing what this PR is meant to fix, right? Could we add such a test?

Reply:
I agree, the variable names in the current test are also misleading. Types defined with "type" and "interface" should be tested separately. Additionally, stubified callbacks for interfaces should be tested—I realized by chance that they weren't stubified before the fix.
I'll update the tests accordingly.

Reply:
Just added new tests for Serializable objects, covering "type" and "interface" definitions individually. Including multi-level objects, objects with stubable callback members, and objects with non-serializable members.
Removed redundant old tests.
Commit: https://github.com/cloudflare/workerd/pull/2040/commits/25cfd1a4ebe1fd33292b0b35fd34ec5215982147